File: modules/mongo/work.js
var mongoose = require('mongoose');

var workSchema = new mongoose.Schema({
    origin_path: {
        type: String,
        required: true,
        lowercase: true,
        index: true
    },
    destination_path: {
        type: String,
        required: true,
        lowercase: true
    },
    user: {
        type: String,
        required: true,
        lowercase: true,
        index: true
    },
    authentication: {
        type: String,
        required: true,
        lowercase: true
    },
    authentication_type: {
        type: String,
        required: true,
        lowercase: true
    },
    command: {
        type: String,
        required: true,
        lowercase: true,
        index: true
    },
    retention: {
        type: Number,
        default: 7
    },
    schedule: {
        type: String,
        default: "* 23 * * * *",
        lowercase: true
    },
    parameters: String,
    server_id: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'server',
        required: true
    },
    deleted: Boolean
}, { collection: 'works' }
);

module.exports = mongoose.model("work", workSchema);

File: axolote_api/api/models/work.py
from django.db import models


class Work(models.Model):

    class Meta:

        db_table = 'work'

    name = models.CharField(max_length=256, unique=True, db_index=True)
    origin_path = models.TextField(
        blank=True,
        null=True
    )
    destination_path = models.TextField(
        blank=True,
        null=True
    )
    command = models.CharField(max_length=256, blank=True, null=True)
    retention = models.CharField(max_length=256, blank=True, null=True)
    schedule = models.CharField(max_length=256, blank=True, null=True)
    parameters = models.CharField(max_length=256, blank=True, null=True)
    deleted = models.BooleanField()
    createddBy = models.CharField(max_length=256, blank=True, null=True)
    description = models.TextField(
        blank=True,
        null=True
    )
    work_type_id = models.ForeignKey(
        'WorkType',
        on_delete=models.PROTECT,
        default=1
    )
    environment_id = models.ForeignKey(
        'Environment',
        on_delete=models.PROTECT,
        default=1
    )
    group_id = models.ForeignKey(
        'Group',
        on_delete=models.PROTECT,
        default=1
    )
    server_id = models.ForeignKey(
        'Server',
        on_delete=models.PROTECT,
        default=1
    )

    def __str__(self):
        return self.name
